tests/kernel/mem_slab/mslab_stats/src/main.c

```diff
@@ -0,0 +1,170 @@
+		      "Expected %zu free bytes, not %zu\n",
+		      BLK_SZ * (NUM_BLOCKS - 1), stats.free_bytes);
+	zassert_equal(stats.allocated_bytes, 1 * BLK_SZ,
+		      "Expected %zu allocated bytes, not %zu\n",
+		      1 * BLK_SZ, stats.allocated_bytes);
+	zassert_equal(stats.max_allocated_bytes, 3 * BLK_SZ,
+		      "Expected %zu max allocated bytes, not %zu\n",
+		      3 * BLK_SZ, stats.max_allocated_bytes);
+
+	/* Allocate 1 block and verify the max is still at 3 */
+
+	status = k_mem_slab_alloc(&kmslab, &memory[1], K_NO_WAIT);
+	zassert_equal(status, 0, "Routine failed with status %d\n", status);
+
+	status = k_mem_slab_runtime_stats_get(&kmslab, &stats);
+	zassert_equal(status, 0, "Routine failed with status %d\n", status);
+
+	zassert_equal(stats.free_bytes, BLK_SZ * (NUM_BLOCKS - 2),
+		      "Expected %zu free bytes, not %zu\n",
+		      BLK_SZ * (NUM_BLOCKS - 2), stats.free_bytes);
+	zassert_equal(stats.allocated_bytes, 2 * BLK_SZ,
+		      "Expected %zu allocated bytes, not %zu\n",
+		      2 * BLK_SZ, stats.allocated_bytes);
+	zassert_equal(stats.max_allocated_bytes, 3 * BLK_SZ,
+		      "Expected %zu max allocated bytes, not %zu\n",
+		      3 * BLK_SZ, stats.max_allocated_bytes);
+
+
+	/* Reset the max allocated blocks; verify max is 2 blocks */
+
+	status = k_mem_slab_runtime_stats_reset_max(&kmslab);
+	zassert_equal(status, 0, "Routine failed with status %d\n", status);
+
+	status = k_mem_slab_runtime_stats_get(&kmslab, &stats);
+	zassert_equal(status, 0, "Routine failed with status %d\n", status);
+
+	zassert_equal(stats.free_bytes, BLK_SZ * (NUM_BLOCKS - 2),
+		      "Expected %zu free bytes, not %zu\n",
+		      BLK_SZ * (NUM_BLOCKS - 2), stats.free_bytes);
+	zassert_equal(stats.allocated_bytes, 2 * BLK_SZ,
+		      "Expected %zu allocated bytes, not %zu\n",
+		      2 * BLK_SZ, stats.allocated_bytes);
+	zassert_equal(stats.max_allocated_bytes, 2 * BLK_SZ,
+		      "Expected %zu max allocated bytes, not %zu\n",
+		      2 * BLK_SZ, stats.max_allocated_bytes);
+
+	/* Free the last two blocks; verify stats results */
+
+	k_mem_slab_free(&kmslab, &memory[0]);
+	k_mem_slab_free(&kmslab, &memory[1]);
+
+	status = k_mem_slab_runtime_stats_get(&kmslab, &stats);
+	zassert_equal(status, 0, "Routine failed with status %d\n", status);
+
+	zassert_equal(stats.free_bytes, BLK_SZ * NUM_BLOCKS,
+		      "Expected %zu free bytes, not %zu\n",
+		      BLK_SZ * NUM_BLOCKS, stats.free_bytes);
+	zassert_equal(stats.allocated_bytes, 0,
+		      "Expected %zu allocated bytes, not %zu\n",
+		      0, stats.allocated_bytes);
+	zassert_equal(stats.max_allocated_bytes, 2 * BLK_SZ,
+		      "Expected %zu max allocated bytes, not %zu\n",
+		      2 * BLK_SZ, stats.max_allocated_bytes);
+}
+
+void test_main(void)
+{
+	ztest_test_suite(lib_mem_slab_stats_test,
+			 ztest_unit_test(test_mem_slab_stats_invalid_params),
+			 ztest_unit_test(test_mem_slab_runtime_stats)
+			 );
+
+	ztest_run_test_suite(lib_mem_slab_stats_test);
```

Comment:
This is crossing with the giant "ZTEST_NEW_API" switchover.  Not a problem for me personally, but I'm sure someone's going to sneak in with a -1 and ask you do use the new stuff.